src/ui/layout/mct-tree.vue

```diff
@@ -556,7 +569,15 @@ export default {
             }
 
             return composition.map((object) => {
-                this.addTreeItemObserver(object, parentObjectPath);
+                // Only add observers and mutables if this is NOT a selector tree
```

Comment:
The main issue I put in this special logic is because I found a case where clicking the "X" to exit out of the create form will not destroy the Locator component itself (note that this doesn't happen if we click "Cancel"). I spent some time trying to fix this and hit a wall, so opted to not track mutables or listeners for this (since we can't change properties of objects while in that modal anyway).

We still have a memory leak there, so we should fix that (I'll make an issue for this if one doesn't exist already). Once that's fixed we can remove this logic and just treat all trees the same in that regard.

Reply:
https://github.com/nasa/openmct/issues/6061

Created issue here. Spent some time today trying to resolve it yet it eludes me still